xl/playlist.py

```diff
@@ -287,44 +287,63 @@ def _iter_uris(pp, tp):
 
-        :param playlist_path: the export path of the playlist
-        :param track_path: the path of the track
+        :param playlist_uri: the export path of the playlist
+        :param track_uri: the path of the track
         :param options: options
         """
-        if options is not None and options.relative:
-            playlist_file = Gio.File.new_for_uri(playlist_path)
-            # Strip playlist filename from export path
-            export_path = playlist_file.get_parent().get_uri()
 
+        track_uri_components = urllib.parse.urlparse(track_uri)
+        export_uri_components = urllib.parse.urlparse(playlist_uri)
+
+        if (
+            export_uri_components.scheme != track_uri_components.scheme
+            or export_uri_components.netloc != track_uri_components.netloc
+        ):
+            # save files to playlist on different location
+            # return track path as is
+            if track_uri_components.scheme == 'file':
+                # as path if local file
+                gio = Gio.File.new_for_uri(track_uri)
+                path = gio.get_path()
+                return Gio.File.new_for_uri(track_uri).get_path()
+            return track_uri
+
+        if options is None or not options.relative:
+            # return absolute uri
+            if track_uri_components.scheme == 'file':
+                # as path if local file
+                return Gio.File.new_for_uri(track_uri).get_path()
+            return track_uri
+
+        # calculate relative path
+        track_uri = Gio.File.new_for_uri(track_uri)
+        playlist_file_folder = Gio.File.new_for_uri(playlist_uri).get_parent()
+
+        if track_uri_components.scheme == 'file':
+            # return path
+            # on Windows relpath raises a ValueError if track and playlist are not on same drive
             try:
-                export_path_components = urllib.parse.urlparse(export_path)
-                track_path_components = urllib.parse.urlparse(track_path)
-            except (AttributeError, ValueError):  # None, empty path
-                pass
-            else:
-                # Only try to retrieve relative paths for tracks with
-                # the same URI scheme and location as the playlist
-                if (
-                    export_path_components.scheme == track_path_components.scheme
-                    and export_path_components.netloc == track_path_components.netloc
-                ):
-                    # Gio.File.get_relative_path does not generate relative paths
-                    # for tracks located above the playlist in the path hierarchy,
-                    # thus process both paths as done here
-                    track_path = os.path.relpath(
-                        track_path_components.path, export_path_components.path
-                    )
+                track_uri = os.path.relpath(
+                    track_uri.get_path(), playlist_file_folder.get_path()
+                )
```

Comment:
The separators are still inconsistent in relative-path variant when running on Windows from source. 

It is not really an issue for most end-users, but it might be confusing when developing/debugging on Windows to see different behavior between source version and PyInstaller-frozen build.

Adding a
```python
                # Ensure '\' is always used as separator on Windows.
                if sys.platform == 'win32' and os.sep == '/':
                    track_uri = track_uri.replace(os.sep, '\\')
```
here should do the trick (requires an `import sys` at the top, though).